code/__DEFINES/stat.dm

```diff
@@ -33,3 +33,8 @@
 #define SHUTTLE_STRANDED 4
 #define SHUTTLE_ESCAPE 5
 #define SHUTTLE_ENDGAME 6
+
+//Movement modes
+#define WALK 0

```

Comment:
I would look into how optimized the BYOND compiler is first. An optimized compiler would see this code:

```C++
m_intent != "walking" // Would be turned from WALK into this during the preprocessor stage.
```

And it would then see that "walking" is constant and would reuse it, never recreating it more than once and never needing to calculate the hash, for equality tests.

In BYOND though? I do not know how it works. If we do assume that this is the case, we could move the \#define into a var/const value. It would be like any other global value and it should be calculated only once.